### DISCUSSION TITLE
Error while running config file

### DISCUSSION THREAD

**Discussion by chargedcs** [qtile/qtile#3532] [Q&A]
# Error while running config file
Hey guys, I recently reloaded my Qtile config, just to make sure everything was alright and realised it wouldn't. I ran the file with python to check for errors and got:
```fish
Traceback (most recent call last):
  File "/home/charged/.config/qtile/config.py", line 113, in <module>
    keys.append(Key([mod], str(i), lazy.group[name].toscreen()))
AttributeError: 'tuple' object has no attribute 'append'
```

The last command I ran before this issue was `paru -Scc`, could this be the issue? Any help is appreciated. [My config](https://gitlab.com/charged1/dotfiles/-/blob/main/.config/qtile/config.py)

**Comment by elParaguayo** [comment]
Get rid of the trailing comma here:
https://gitlab.com/charged1/dotfiles/-/blob/main/.config/qtile/config.py#L90

**Reply by chargedcs** [reply]
That didn't work, and I can't open my X11 display now...

**Reply by elParaguayo** [reply]
Then there's something else wrong with your config. The tuple issue is fixed with that change:
```
>>> a=[1,2],
>>> b=[1,2]
>>> type(a)
<class 'tuple'>
>>> type(b)
<class 'list'>
```
What else have you changed as this has nothing to do with your `paru` command?

**Reply by SweepingBishops** [reply]
What error do you get now?

**Reply by chargedcs** [reply]
@elParaguayo I've been having lots of Arch issues lately, so I installed Arco Linux, and linked up my dotfiles and everything. **The config works when run with `python3`, but Qtile won't launch at all.**

**Reply by SweepingBishops** [reply]
Can you please share the logs?

**Reply by chargedcs** [reply]
Which logs would you like?

**Reply by SweepingBishops** [reply]
~/.local/share/qtile/qtile.log

**Reply by SweepingBishops** [reply]
Preferably just after a failed launch.

**Reply by chargedcs** [reply]
Hold on, I think I found a bunch of errors myself...

**Reply by chargedcs** [reply]
@elParaguayo @SweepingBishops thanks for your help, the logs showed me the error and I fixed it :)
